backend/app/api/endpoints/openapi_responses.py

```
@@ -592,58 +530,81 @@ def _query_subtasks():
+            except _DispatchWithoutTerminalError as e:
+                await _persist_terminal_failure(
+                    subtask_id=assistant_subtask_id,
+                    task_id=task_kind_id,
+                    error_message=str(e),
+                )
+                logger.exception(f"[BACKGROUND] Error in background task: {e}")
             except Exception as e:
                 logger.exception(f"[BACKGROUND] Error in background task: {e}")
-                await _handle_failure(str(e))
 
         asyncio.create_task(_run_background_task())
```

Ruff (0.15.12)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)